tools/functional-tester/etcd-tester/checks.go

```diff
@@ -133,7 +133,8 @@ func (lc *leaseChecker) checkShortLivedLease(ctx context.Context, leaseID int64)
 	var resp *pb.LeaseTimeToLiveResponse
 	for i := 0; i < retries; i++ {
 		resp, err = lc.getLeaseByID(ctx, leaseID)
-		if rpctypes.Error(err) == rpctypes.ErrLeaseNotFound {
+		// lease not found when it's follower or leader
+		if (err == nil && resp.TTL == -1) || (err != nil && rpctypes.Error(err) == rpctypes.ErrLeaseNotFound) {
```

Comment:
was mostly concerned about compat with older clusters, but I guess it only returned ErrLeaseNotFound in the rc's for 3.1?